src/components/Resource/Forms/Training/TrainingSubmit.tsx

```diff
@@ -61,6 +62,17 @@ type TrainingSettingsType = {
   };
 };
 
+const getPrecision = (n: any) => {
+  if (!isFinite(n)) return 0;
+  const e = 1;
+  let p = 0;
+  while (Math.round(n * e) / e !== n) {
+    n *= 10;
+    p++;
+  }
+  return p;
+};
```

Comment:
Not sure what this does, but I trust you on this one 😛 maybe add comments for future references?

Reply:
lol it just takes a number and computes how many decimals points there are. i can add a comment to it